src/engine/sys/sdl_glimp.cpp

```diff
@@ -2285,77 +2286,84 @@ static void GLimp_InitExtensions()
+		// which results in either glitches or the shader compiler crashing (when material system is enabled)
+		if ( bindlessTextureEnabled && glConfig2.driverVendor == glDriverVendor_t::MESA )
 		{
-			std::string mesaVersion = "";
+			if ( workaround_glDriver_mesa_v241_disableBindlessTexture.Get() )
+			{
+				bool foundMesa241 = false;
 
-			static const std::vector<std::pair<std::string, std::vector<std::string>>> versions = {
-				{ "1.0", { "1.0-devel", "1.0-rc1", "1.0-rc2", "1.0-rc3", "1.0-rc4", "1.0", } },
-				{ "1.",  { "1.1", "1.2", "1.3", "1.4", } },
-				{ "2.0", { "2.0-devel", "2.0-rc1", } },
-			};
+				std::string mesaVersion = "";
+
+				static const std::pair<std::string, std::vector<std::string>> versions[] = {
+					{ "1.0", { "1.0-devel", "1.0-rc1", "1.0-rc2", "1.0-rc3", "1.0-rc4", "1.0", } },
+					{ "1.",  { "1.1", "1.2", "1.3", "1.4", } },
+					{ "2.0", { "2.0-devel", "2.0-rc1", } },
+				};
 
-			for ( auto& vp : versions ) {
-				mesaVersion = Str::Format( "Mesa 24.%s", vp.first );
+				for ( auto& vp : versions ) {
+					mesaVersion = Str::Format( "Mesa 24.%s", vp.first );
 
-				if ( Q_stristr( glConfig.version_string, mesaVersion.c_str() ) ) {
-					for ( auto& v : vp.second ) {
-						mesaVersion = Str::Format( "Mesa 24.%s", v );
+					if ( Q_stristr( glConfig.version_string, mesaVersion.c_str() ) ) {
+						for ( auto& v : vp.second ) {
+							mesaVersion = Str::Format( "Mesa 24.%s", v );
 
-						if ( Q_stristr( glConfig.version_string, mesaVersion.c_str() ) ) {
-							foundMesa241 = true;
-							break;
+							if ( Q_stristr( glConfig.version_string, mesaVersion.c_str() ) ) {
+								foundMesa241 = true;
+								break;
+							}
 						}
-					}
 
-					break;
+						break;
+					}
 				}
-			}
 
-			if ( foundMesa241 ) {
-				logger.Warn( "...found buggy %s driver, ARB_bindless_texture disabled", mesaVersion );
+				if ( foundMesa241 ) {
+					logger.Warn( "Found buggy %s driver, disabling ARB_bindless_texture.", mesaVersion );
+					bindlessTextureEnabled = false;
+				}
 			}
-		}
-
-		bool foundOglp = false;
 
-		if ( workaround_glDriver_amd_oglp_disableBindlessTexture.Get() )
-		{
 			/* AMD OGLP driver for Linux shares the same vendor string than AMD Adrenalin driver
 			for Windows and AMD ATI driver for macOS. When running the Windows engine binary on
 			Wine we must check we're not running Windows or macOS to detect Linux OGLP. */
-			if ( Q_stristr( glConfig.vendor_string, "ATI Technologies Inc." ) )
+			if ( workaround_glDriver_amd_oglp_disableBindlessTexture.Get() )
 			{
-				#if defined(_WIN32)
-					// Detect Wine being used.
-					if ( Sys::isRunningOnWine() )
-					{
-						// Detect AMD ATI driver on macOS not being used.
-						if ( !Q_stristr( glConfig.version_string, "ATI-" ) )
+				bool foundOglp = false;
+
+				if ( glConfig2.hardwareVendor == glHardwareVendor_t::ATI
+					&& glConfig2.driverVendor != glDriverVendor_t::MESA )
+				{
+					#if defined(_WIN32)
```

Comment:
I've received a report that it's also not working on the windows AMD proprietary driver (which is the Adrenalin one I guess).

Reply:
Apparently not Adrenalin, but I don't think that changes anything here, it should still be disabled on non-Mesa ATI, regardless of if running in Wine.

Reply:
> Apparently not Adrenalin

What does that mean? Does that mean the AMD proprietary driver is not Adrenalin? or that Adrenalin is not buggy?

Is there a difference between Windows provided drivers and separate download from AMD?

Reply:
No idea, I was told that Adrenain had to be uninstalled because it was apparently hard-locking in unv after some Adrenalin update, and that they're using the generic AMD driver. I ask to clarify later.

Reply:
I assume Windows stock driver is just older Adrenalin, there is no reason to have two drivers there.

Reply:
I split the workaround cvar between OGLP and Adrenalin because there is strong reasons to believe the current OGLP driver is the last one and will never be updated. While AMD has no Adrenalin alternative on Windows and then is very likely to update it.

But I make no difference between AMD stock driver on Windows and self-installed Adrenalin driver (I name them both Adrenalin).

Reply:
Yeah, probably. To me the description they have on their website reads like Adrenalin is just a companion app like Geforce Experience, that makes getting a driver update faster, changing some setting etc.

Anyway, this should work fine.